### PR TITLE
fix: Left Sidebar Not moving down

### LINKED ISSUE #6534: Public Event Page: Left Sidebar not longer moving down on scroll down
One of the recent changes changed the behavior of the left sidebar. The left sidebar used to move down when the user scrolls down the page. This behavior disappeared. Please re-instate this sidebar behavior.

Compare https://eventyay.com/e/fa96ae2c/

Screenshot from 2021-01-29 20-50-07

### PR DESCRIPTION
**Pull request by dhruvin1831**
Fixes #6534

#### Short description of what this resolves:
The left sidebar was not sticky which was due to a parent container with class "footer-pusher" having overflow-x as hidden.

#### Changes proposed in this pull request:
I have removed this property of the above class as it is already inherited from the body of the page. This fixes the bug and no functionality/UI is hampered.
-
-
-
Screenshot 2021-02-02 155436

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/styles/partials/footer.scss [hunk 1]
```diff
@@ -24,15 +23,16 @@ footer {
   padding-top: 30px !important;
 
   .language.menu > div {
-    color: rgba(0, 0, 0, .87) !important;
+    color: rgba(0, 0, 0, 0.87) !important;
```
Comment (iamareebjamal) [inline app/styles/partials/footer.scss:26]: Why

File: app/styles/partials/footer.scss [hunk 2]
```diff
@@ -24,15 +23,16 @@ footer {
...
   }
 }
 
 .language.menu {
   min-height: 400px !important;
 }
 
-@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) { // sass-lint:disable-line no-vendor-prefixes
+@media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
```
Comment (iamareebjamal) [inline app/styles/partials/footer.scss:34]: Why?

**Comment by iamareebjamal** [comment]
This reintroduces https://github.com/fossasia/open-event-frontend/issues/6194

File: app/styles/partials/footer.scss (hunk 1 above)
Comment (maze-runnar) [inline app/styles/partials/footer.scss:26]: revert

Comment (dhruvin1831) [reply app/styles/partials/footer.scss:26]: These changes occurred automatically when I made a commit after changing the .scss file - Error message"We've failed to pass the specified git pre-commit hooks as the `lint` pre-commit: hook returned an exit code (1)".

It further said that it cannot be 0.

Please help me solve this issue.

File: app/styles/partials/footer.scss (hunk 2 above)
Comment (dhruvin1831) [reply app/styles/partials/footer.scss:34]: These changes occurred automatically when I made a commit after changing the .scss file - Error message"We've failed to pass the specified git pre-commit hooks as the lint pre-commit: hook returned an exit code (1)".

It further said that it cannot be 0.

Please help me solve this issue.

File: app/styles/partials/footer.scss (hunk 1 above)
Comment (dhruvin1831) [reply app/styles/partials/footer.scss:26]: I'm on it

**Comment by dhruvin1831** [comment]
@maze-runnar @iamareebjamal I have reverted the unnecessary changes spotted by you and fixed it.

I am now working/finding the solution for #6194.

Removing Overflow-x is necessary for the sticky behavior of the child element.

**Comment by Creatoon** [comment]
@dhruvin1831 I found , twitter's iframe min-width, paddings and margin on the main-container and also .ui.relaxed.grid on public-event class is the reason for overflowing nature of footer-pusher, if you will fix this , overflowing nature will be fixed.

**Comment by iamareebjamal** [comment]
Not fixed

**Comment by dhruvin1831** [comment]
@iamareebjamal  I am not able to reproduce the bug on my side and it's working fine. Please help me to reproduce it or any other thing I need to fix.

**Comment by iamareebjamal** [comment]
Go to event page and change browser to responsive mobile mode

**Comment by Creatoon** [comment]
> @iamareebjamal I am not able to reproduce the bug on my side and it's working fine. Please help me to reproduce it or any other thing I need to fix.

You will only be able to see change when you turn off the overflow from footer pusher and then try to figure out what i asked you to.

**Comment by dhruvin1831** [comment]
> > @iamareebjamal I am not able to reproduce the bug on my side and it's working fine. Please help me to reproduce it or any other thing I need to fix.
> 
> You will only be able to see change when you turn off the overflow from footer pusher and then try to figure out what i asked you to.

@Creatoon I have already turned off the overflow from footer pusher in previous commits.

**Comment by dhruvin1831** [comment]
@iamareebjamal 
https://user-images.githubusercontent.com/70140023/106867714-f1845600-66f3-11eb-99b7-773675891e12.mp4

#

File: app/styles/partials/footer.scss [hunk 3]
```diff
@@ -1,10 +1,15 @@
 .footer-pusher {
   display: flex;
   flex-direction: column;
-  overflow-x: hidden;
   min-height: 100vh;
 }
 
+#main-container {  // sass-lint:disable-line no-ids
```
Comment (iamareebjamal) [inline app/styles/partials/footer.scss:7]: Why this?

File: app/templates/public.hbs [hunk 4]
```diff
@@ -44,7 +44,7 @@
         {{outlet}}
       </div>
       {{/unless}}
-      <div class="three wide column">
+      <div class="three wide column" style="overflow-x: hidden;">
```
Comment (iamareebjamal) [inline app/templates/public.hbs:47]: Why not add a class?

File: app/styles/partials/footer.scss (hunk 3 above)
Comment (dhruvin1831) [reply app/styles/partials/footer.scss:7]: It is necessary due to the precedence of styling. There are other classes which are overriding the properties:

_**@media only screen and (min-width: 768px) and (max-width: 991px)
.ui.container {
    width: 723px;
    margin-left: auto!important;
    margin-right: auto!important;_**

Hence to override this id selector is the way. There is already a class named main-container but its been overridden by the media query and class .ui.container.

File: app/templates/public.hbs (hunk 4 above)
Comment (dhruvin1831) [reply app/templates/public.hbs:47]: Please suggest if there is an existing class or I need to add a new one.

File: app/styles/partials/footer.scss (hunk 3 above)
Comment (dhruvin1831) [reply app/styles/partials/footer.scss:7]: @media only screen and (max-width: 767px)
.ui.container {
    margin-left: 1em!important;
    margin-right: 1em!important;
    
    Also, this media query is overriding the .main-container class for small screen devices.

File: app/templates/public.hbs (hunk 4 above)
Comment (dhruvin1831) [reply app/templates/public.hbs:47]: I think adding a new class is the way. I am working on this.

File: app/styles/partials/footer.scss (hunk 3 above)
Comment (iamareebjamal) [reply app/styles/partials/footer.scss:7]: But why is this style necessary. Won't it break other pages?

Comment (dhruvin1831) [reply app/styles/partials/footer.scss:7]: The parent container footer-pusher was overflowed by the main-container. I think Adding/overriding default margins (added by UI.container) will only prevent overflow but the parent container's dimensions will not be affected if used in other pages.

Comment (dhruvin1831) [reply app/styles/partials/footer.scss:7]: Other pages (if they have any of these containers) will still have the default .ui.containers properties they are not overridden.

File: app/templates/public.hbs (hunk 4 above)
Comment (dhruvin1831) [reply app/templates/public.hbs]: I have added a new commit to solve this [https://github.com/fossasia/open-event-frontend/pull/6582/commits/953b8a68082ea7fed713fbdfe55eeb642e2f9e76](url)

Comment (dhruvin1831) [reply app/templates/public.hbs]: I have added a new commit to solve this. [https://github.com/fossasia/open-event-frontend/pull/6582/commits/953b8a68082ea7fed713fbdfe55eeb642e2f9e76](url)

Comment (dhruvin1831) [reply app/templates/public.hbs]: I have added a new commit to solve this issue.

**Comment by iamareebjamal** [comment]
@dhruvin1831 As expected, it causes #6603

**Comment by dhruvin1831** [comment]
@iamareebjamal I am working on another possible solution.

**Pull request merged**